app/jobs/socure_docv_results_job.rb

```diff
@@ -3,41 +3,76 @@
 class SocureDocvResultsJob < ApplicationJob
   queue_as :default
 
-  attr_reader :document_capture_session_uuid
+  attr_reader :document_capture_session_uuid, :async
 
   # @param [String] document_capture_session_uuid
   def perform(document_capture_session_uuid:)
     @document_capture_session_uuid = document_capture_session_uuid
+    @async = IdentityConfig.store.ruby_workers_idv_enabled
 
-    dcs = DocumentCaptureSession.find_by(uuid: document_capture_session_uuid)
-    raise "DocumentCaptureSession not found: #{document_capture_session_uuid}" if !dcs
+    raise "DocumentCaptureSession not found: #{document_capture_session_uuid}" unless
+      document_capture_session
 
-    # analytics = create_analytics(
-    #   user: dcs.user,
-    #   service_provider_issuer: dcs.issuer,
-    # )
-
-    result = socure_document_verification_result
-    dcs.store_result_from_response(result)
+    timer = JobHelpers::Timer.new
+    response = timer.time('vendor_request') do
+      socure_document_verification_result
+    end
+    log_verification_request(
+      docv_result_response: response,
+      vendor_request_time_in_ms: timer.results['vendor_request'],
+    )
+    document_capture_session.store_result_from_response(response)
   end
 
   private
 
-  def create_analytics(
-    user:,
-    service_provider_issuer:
-  )
-    Analytics.new(
-      user:,
-      request: nil,
-      sp: service_provider_issuer,
-      session: {},
+  def analytics
+    @analytics ||= Analytics.new(
+      user: document_capture_session.user,
+      service_provider_issuer: document_capture_session.issuer,
     )
   end
 
+  def log_verification_request(docv_result_response:, vendor_request_time_in_ms:)
+    return if docv_result_response.nil?
+
+    analytics.idv_socure_verification_data_requested(
+      **filtered_verification_data(docv_result_response).merge(
+        docv_transaction_token: document_capture_session.socure_docv_transaction_token,
+        submit_attempts: rate_limiter&.attempts,
+        remaining_submit_attempts: rate_limiter&.remaining_count,
+        vendor_request_time_in_ms:,
+        async:,
+      ).compact,
+    )
+  end
+
+  def filtered_verification_data(docv_result_response)
+    hash = docv_result_response.to_h
+    hash.delete(:attention_with_barcode)
+    hash.delete(:selfie_live)
+    hash.delete(:selfie_quality_good)
+    hash.delete(:selfie_status)
+    hash
+  end
```

Comment:
```suggestion
    analytics.idv_socure_verification_data_requested(
      **filtered_verification_data(docv_result_response).merge(
        docv_transaction_token: document_capture_session.socure_docv_transaction_token,
        submit_attempts: rate_limiter&.attempts,
        remaining_submit_attempts: rate_limiter&.remaining_count,
        vendor_request_time_in_ms:,
        async:,
      ).except(:attention_with_barcode, :selfie_live, :selfie_quality_good, :selfie_status).compact,
    )
  end
```

Reply:
Regarding the removing fields from the hash. In a separate PR, this could be a great opportunity for refactoring.  Most of the fields you are trying to discard are very specific to TrueID and could possibly be refactored into the TrueIDResponse 🤔

Reply:
Good idea, will ticket

Reply:
Ticketed: https://cm-jira.usa.gov/browse/LG-15175